examples/Atlas/+atlasUtil/propertyCache.m

```diff
@@ -0,0 +1,46 @@
+function prop_cache = propertyCache(r)
+% NOTEST
+% Functions like findLinkId, getTerrainContactPoints, etc. can be too slow to call
+% in the inner loop of our controller or planner, so we cache some useful information
+% at setup time. 
+% @param r an Atlas
+
+typecheck(r, 'Atlas');
+prop_cache = struct('contact_groups', [],...
+                    'body_ids', struct(),...
+                    'position_indices', struct(),...
+                    'actuated_indices', [],...
+                    'nq', 0,...
+                    'nv', 0,...
+                    'num_bodies', 0);
+
+% getTerrainContactPoints is pretty expensive, so we'll just call it
+% for all the bodies and cache the results
+nbod = length(r.getManipulator().body);
+contact_group_cache = cell(1, nbod);
+for j = 1:nbod
+  contact_group_cache{j} = struct();
+  for f = 1:length(r.getBody(j).collision_geometry_group_names)
+    name = r.getBody(j).collision_geometry_group_names{f};
+    contact_group_cache{j}.(name) = r.getBody(j).getTerrainContactPoints(name);
+  end
+end
+
+prop_cache.contact_groups = contact_group_cache;
+
+prop_cache.nq = r.getNumPositions();
```

Comment:
i can see findLinkId, etc.  but hopefully getNumPositions and getNumVelocities is no slower....?